RELEASE.md

```diff
@@ -4,7 +4,7 @@ The Kubernetes Template Project is released on an as-needed basis. The process i
 
 1. An issue is proposing a new release with a changelog since the last release
 2. All [OWNERS](OWNERS) must LGTM this release
-3. Bump the `version` and `appVersion` of the Helm chart in `charts/karpenter/Chart.yaml`
-4. An OWNER runs `git tag -s $VERSION` and inserts the changelog and pushes the tag with `git push $VERSION`
+3. An OWNER runs `git tag -s $VERSION` and inserts the changelog and pushes the tag with `git push $VERSION`
```

Comment:
i think we should keep the line about bumping the version and appVersion of the helm chart.